### ISSUE TITLE
Allow Kubernetes TLS Secrets with CA certificate only

### ISSUE THREAD

**Issue by stephan2012** [enix/x509-certificate-exporter#20]
In some of our namespaces, Kubernetes TLS Secrets exist that hold a CA certificate only, i.e., the key `ca.crt` is not empty while `tls.crt` and `tls.key` are. This raises warnings for the X.509 Certificate Exporter and an error count in metric `x509_read_errors`:

```text
time="2021-10-29T15:57:48Z" level=warning msg="no certificate(s) found in \"k8s/grafana/repo-tls-cert\""
time="2021-10-29T15:57:48Z" level=warning msg="no certificate(s) found in \"k8s/log/repo-tls-cert\""
```

Sample secret:

```yaml
apiVersion: v1
kind: Secret
type: kubernetes.io/tls
metadata:
  creationTimestamp: "2021-04-20T07:20:35Z"
  name: repo-tls-cert
  namespace: log
  resourceVersion: "170394169"
  selfLink: /api/v1/namespaces/log/secrets/repo-tls-cert
  uid: 00cdec3c-4f30-47cc-9fcb-83a21edeb81d
data:
  ca.crt: LS...
  tls.crt: ""
  tls.key: ""
```

It would be nice to have a switch that allows CA-certificate-only secrets that do not raise warnings. For me, only a Kubernetes Secret that holds neither `ca.crt` nor `tls.crt` is something to worry about.

**Comment by npdgm** [comment]
Hi Stephan,
That's a really good point, we didn't come across this situation so far. Thanks for reporting your use case.
We'll implement support for the CA shortly
Cheers

**Comment by stephenharris** [comment]
I came across this too, but with a secret that only contained `tls.crt`. The helm chart by default as the following:

```
    secretTypes:
    - type: kubernetes.io/tls
      key: tls.crt
    - type: kubernetes.io/tls
      key: ca.crt
```

which means it will check each secret of `kubernetes.io/tls` for [both](https://github.com/enix/x509-certificate-exporter/blob/e0b8ea13970a76b3b765b13cc62ced9fa9ec14bb/internal/kubernetes.go#L46) tls.crt and ca.crt. If it fails to find one or the other, it will emit a warning and increment `x509_read_errors` metric. The one it does find it will emit the desired metrics.

It would be great if, for a secret of a given type, it checked each listed key and only warned if none were found. However, that's a breaking change.

**Comment by npdgm** [comment]
Hi @stephenharris 
Oh. You are right, while working on this original issue we introduced this regression. I wasn't quite satisfied with the behavior and that's why no news was given here yet. But as you pointed out it's actually worse and calls for a revert.
Hence I just rolled back the changes, a new version of the Helm chart is available, version `1.20.0`. Which is the same as running `1.18.0` but I want to prevent anyone installing the bad one :disappointed: 
So we're going back to the drawing board with @arcln. This simple feature has to be implemented an other way, generic code gets in the way of handling this properly without error.
Sorry about the troubles,
Cheers
